lib/openstudio-standards/standards/ashrae_90_1_prm/ashrae_90_1_prm.Model.rb

```diff
@@ -2024,32 +2110,35 @@ def model_update_ground_temperature_profile(model, climate_zone)
       end
     end
 
-    return false unless surfaces_with_fc_factor_boundary
+    if surfaces_with_fc_factor_boundary
+      # Remove existing FCFactor temperature profile
+      model.getSiteGroundTemperatureFCfactorMethod.remove
 
-    # Remove existing FCFactor temperature profile
-    model.getSiteGroundTemperatureFCfactorMethod.remove
+      # Get path to weather file specified in the model
+      weather_file_path = prm_get_optional_handler(model.getWeatherFile, @sizing_run_dir, 'path').to_s
 
-    # Get path to weather file specified in the model
-    weather_file_path = model.getWeatherFile.path.get.to_s
-
-    # Look for stat file corresponding to the weather file
-    stat_file_path = weather_file_path.sub('.epw', '.stat').to_s
-    if !File.exist? stat_file_path
-      # When the stat file corresponding with the weather file in the model is missing,
-      # use the weather file that represent the climate zone
-      climate_zone_weather_file_map = model_get_climate_zone_weather_file_map
-      weather_file = climate_zone_weather_file_map[climate_zone]
-      stat_file_path = model_get_weather_file(weather_file).sub('.epw', '.stat').to_s
-    end
+      # Look for stat file corresponding to the weather file
+      stat_file_path = weather_file_path.sub('.epw', '.stat').to_s
+      if !File.exist? stat_file_path
+        # When the stat file corresponding with the weather file in the model is missing,
+        # use the weather file that represent the climate zone
+        climate_zone_weather_file_map = model_get_climate_zone_weather_file_map
+        prm_raise(climate_zone_weather_file_map.key?(climate_zone),
+                  @sizing_run_dir,
+                  "Failed to find a matching climate zone #{climate_zone} from the climate zone weather files.")
+        weather_file = climate_zone_weather_file_map[climate_zone]
+        stat_file_path = model_get_weather_file(weather_file).sub('.epw', '.stat').to_s
+      end
 
-    ground_temp = OpenStudio::Model::SiteGroundTemperatureFCfactorMethod.new(model)
-    ground_temperatures = model_get_monthly_ground_temps_from_stat_file(stat_file_path)
-    unless ground_temperatures.empty?
-      # set the site ground temperature building surface
-      ground_temp.setAllMonthlyTemperatures(ground_temperatures)
+      ground_temp = OpenStudio::Model::SiteGroundTemperatureFCfactorMethod.new(model)
+      ground_temperatures = model_get_monthly_ground_temps_from_stat_file(stat_file_path)
+      unless ground_temperatures.empty?
+        # set the site ground temperature building surface
+        ground_temp.setAllMonthlyTemperatures(ground_temperatures)
+      end
     end
 
-    return true
+    return surfaces_with_fc_factor_boundary
```

Comment:
update return type in method documetnation

Reply:
The `surfaces_with_fc_factor_boundary` is a Boolean in the code.
I updated the comment to:
`# @return [Boolean] surfaces_with_fc_factor_boundary,  returns true if successful, false if not`